ray-operator/apis/ray/v1/rayservice_types.go

```diff
@@ -22,6 +22,9 @@ const (
 type RayServiceUpgradeType string
 
 const (
+	// During upgrade, IncrementalUpgrade strategy will create an upgraded cluster to gradually scale
+	// and migrate traffic to using Gateway API.
+	IncrementalUpgrade RayServiceUpgradeType = "IncrementalUpgrade"
```

Comment:
If we're still nit picking on the naming here, I think the new strategy type shoud be "NewClusterWithUpgrade` or `NewClusterWithIncrementalUpgrade`. This makes it more obvious that we are still upgrading a new cluster. `IncrementalUpgrade` makes it ambigious if the existing cluster or new cluster is being upgraded